### DISCUSSION TITLE
Is there a non-tedious way of creating 81 text boxes?

### DISCUSSION THREAD

**Discussion by abourque72** [hecrj/iced#1377] [Q&A]
As the title suggests, I want to create 81 text boxes. I am making a GUI for a sudoku solver. So far I have enjoyed the simplicity of Iced, but I understand if I may need to switch libraries in order to achieve something like this. My current issue is that it looks like I need to implement 81 different Messages in my enum.  I think I can use a loop to make all the TextInputs, but I believe each would require its own Message. If this is not correct, I would love to know.

Thanks

**Comment by Kaiden42** [comment]
You can use the same message for each textinput. You could, for example, store the xy-Coordinates in the message.

```
enum Message {
    FieldInput(u8, u8, String)
}

...

text_input(
"Placeholder",
 input_value,
 move |value| Message::FieldInput(x, y, value),
)
```
(I've written this from memory. I've not tested this as I'm currently not on my PC)

The todos example shows a similar aproach.
https://github.com/iced-rs/iced/blob/e6e3eff8762e9e8350f00b340348dc2261dd0053/examples/pure/todos/src/main.rs#L43
https://github.com/iced-rs/iced/blob/e6e3eff8762e9e8350f00b340348dc2261dd0053/examples/pure/todos/src/main.rs#L179

**Reply by abourque72** [reply]
Thanks! I am now having trouble initializing the textboxes. I am trying to use arrays since I have a static number of them, but I'm not sure how to initialize the array at first, and then when I try to go through and set them, I get an error saying "`self.inputs[_][_]` was mutably borrowed here in the previous iteration of the loop"
I suppose I could use a vector to add them one by one so I don't have to deal with initialization of the array, but I think I would still get that error.
I'm also confused on the difference between iced and iced::pure. The example you linked uses pure functions, and text_input is one as well. What is the difference?

Thanks again!

**Reply by Kaiden42** [reply]
If you try to mutably access a collection using `foo[i]`, it is just a shorthand for writing `foo.get_mut(i)`. Following the [signature](https://doc.rust-lang.org/stable/std/primitive.slice.html#method.get_mut) of the function, you'll see that you will mutably borrow the whole collection, while you are holding the mutable borrow of the element `i`. Therefore you can't borrow the collection more than once. For borrowing all elements of a collection mutably you will have to use the iterator of the collection like `foo.iter_mut()`.

Iced pure is the new and recommended way of using iced (unless you want to try to manipulate the internal states of the widgets, but that usecase is really rare). With the pure feature you don't have to deal with the internal states of the widgets anymore. Additionally it removes a lot of boilerplate to by giving you some helper functions for creating your elements.

I've created two gist that will show you how to solve your problem with either iced native or iced pure.

Iced native:
https://gist.github.com/Kaiden42/07581c9c1c55b9b18d08c83b821657c8

iced pure:
https://gist.github.com/Kaiden42/4ffbe023ec8568b52a6f0aee14d5a0b2
